daffodil-japi/src/main/scala/org/apache/daffodil/japi/Daffodil.scala

```diff
@@ -373,6 +374,12 @@ class ProcessorFactory private[japi] (private var pf: SProcessorFactory)
     new DataProcessor(dp)
   }
 
+  /**
+   * Exposes the RootView object corresponding to this ProcessorFactory.  This can
+   * be used to start a walk using the walkFromRoot method in a DSOM Walker.
+   */
+  def rootView: RootView = pf.rootView
```

Comment:
We probably should document that this is an experiemental API and is subject to change. Once we stabalize this and confirm we have the right functionality, we can remove the experiemtal tag and make the Views part of the public API.

Reply:
What is the preferred way to mark an API as experimental in Daffodil?  Is there an Annotation or ScalaDoc tag, or should I simply state it's Experimental in the comment describing `rootView`?

Reply:
I know the Scala language sometimes uses an ``experimental`` object that things go in, so we could do something like that:
```scala
class ProcoessorFactory(pf: SProcessorFactory) {
  ...
  object experimental {
    def rootView: RootView = pf.rootView
  }
  ...
}
```
To access the rootView, one would do something like this:
```scala
val rootView = pf.experimental.rootView
```
That makes it very obvious when you are using something that is experimental and that the API might change in the future. If we do go this route, I wonder if it also make sense to create a new sbt subproject and move all the new View traits and AbstratDSOMWalker in an experimental package (eg. org.apache.daffodil.experimental.dsom.walker), similar to how we have the daffodil-udf package split out? And we can use the new unidoc stuff to include it in public API docs?

Not sure how I feel about this though. Maybe it's too obtrusive?

I think the other option is to just have a blurb like "This API is experimental is and subject to change". I don't know if there's an annotation or standard javadoc practice for this kind of thing.

Reply:
I think a separate submodule is going too far unless that's necessary someday to grab the scaladoc from this and incorporate it into the public-facing scaladoc set. 
I like the experimental object technique you showed. That's sufficient to me for now.

Reply:
Yeah, the submodule is really only needed when we want to include View stuff in public docs. It does mean in order to use the walker/views you kindof have to know what you're doing and know where to find these views. But that's not unreasonable for experimental things. Agree that the subprojet probably isn't needed right now.

Reply:
While this solution is great and simple for Scala, it doesn't seem to translate well to Java; `pf.experimental.rootView()` does not compile, and IntelliJ thinks that `ProcessorFactory.experimental$.MODULE$.rootView()` compiles but this does not work when you go to build it, either.  Could we just create a brief inner wrapper class here with its only member as `rootView` and then make `experimental` a member of `ProcessorFactory` of that wrapper type?

Reply:
Seems like a reasonable approach to me.